nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/CopyS3Object.java

```diff
@@ -119,28 +145,127 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         final String destinationBucket = context.getProperty(DESTINATION_BUCKET).evaluateAttributeExpressions(flowFile).getValue();
         final String destinationKey = context.getProperty(DESTINATION_KEY).evaluateAttributeExpressions(flowFile).getValue();
 
+        final GetObjectMetadataRequest sourceMetadataRequest = new GetObjectMetadataRequest(sourceBucket, sourceKey);
+        final ObjectMetadata metadataResult = s3.getObjectMetadata(sourceMetadataRequest);
+
+        if (metadataResult == null) {
+            session.transfer(flowFile, REL_FAILURE);
+        }
+
+        final AtomicReference<String> multipartIdRef = new AtomicReference<>();
+        final long contentLength = metadataResult.getContentLength();
+        final boolean isMultiPart = metadataResult.getContentLength() > MULTIPART_THRESHOLD;
```

Comment:
Recommend renaming this:
```suggestion
        final boolean multipartUploadRequired = metadataResult.getContentLength() > MULTIPART_THRESHOLD;
```